tests/unit/valgrind.suppressions

```diff
@@ -1,13 +1,48 @@
-# It looks like valgrind may generate false positives on pthreads: https://stackoverflow.com/a/13132968
+# Valgrind may generate false positives on pthreads: https://stackoverflow.com/a/13132968
+# Without these suppressions, the following tests will fail:
+# s2n_examples_test, s2n_fork_generation_number_test, s2n_init_test, s2n_key_update_threads_test, and s2n_random_test.
 {
-   pthred_false_positive
+   pthread_false_positive
    Memcheck:Leak
    match-leak-kinds: possible
    fun:calloc
+   ...
    fun:allocate_dtv
    fun:_dl_allocate_tls
    fun:allocate_stack
-   fun:pthread_create@@GLIBC_2.2.5
+   fun:pthread_create@@*
+   ...
+   fun:main
+}
+
+# https://man7.org/linux/man-pages/man3/backtrace_symbols_fd.3.html
+#  backtrace() and backtrace_symbols_fd() don't call malloc()
+#  explicitly, but they are part of libgcc, which gets loaded
+#  dynamically when first used.  Dynamic loading usually triggers
+#  a call to [malloc(3)](https://man7.org/linux/man-pages/man3/malloc.3.html).
```

Comment:
As [discussed](https://github.com/aws/s2n-tls/pull/4776#discussion_r1768119427), this comment doesn't actually explain the memory leak. Rather than just quoting the backtrace man page, can you briefly explain in the comment why the memory leak is not considered an issue? I wonder if linking to the PR comment thread with your testing would be too strange 🤔

Also, nits: You probably want to use more than one leading space to indicate a quote-- it's pretty subtle right now. And I don't think the markdown formatting helps.

Reply:
> can you briefly explain in the comment why the memory leak is not considered an issue? I wonder if linking to the PR comment thread with your testing would be too strange.

I add a brief explanation for such suppression. If people have questions about this, they can view the git blame to find this PR. I will add my PR comment's link to the call-out section of PR descriptions to help them locate the full explanation.

>  Also, nits: You probably want to use more than one leading space to indicate a quote-- it's pretty subtle right now. And I don't think the markdown formatting helps.

This comment no longer contains any quotations, but I will keep this in mind for the future.